robot/src/main/java/us/ilite/robot/modules/Elevator.java

```diff
@@ -0,0 +1,280 @@
+
+        //Create default NEO and set the ramp rate
+        mMasterElevator = SparkMaxFactory.createDefaultSparkMax(kCansparkId, MotorType.kBrushless);
+        mMasterElevator.setIdleMode(IdleMode.kBrake);
+        mMasterElevator.setRampRate(SystemSettings.kELevatorControlLoopPeriod);
+
+
+        //We start at the bottom
+        mAtBottom = true;
+        mAtTop = false;
+        mDesiredDirectionUp = true;
+
+        //Make sure the elevator is stopped upon initialization
+        mDesiredPosition = EElevatorPosition.BOTTOM;
+        mCurrentState = EElevatorState.STOP;
+
+        mCurrentEncoderTicks = 0;
+    }
+
+    public void shutdown(double pNow) {
+
+    }
+
+    public void modeInit(double pNow) {
+
+    }
+
+    public void periodicInput(double pNow) {
+
+    }
+
+    public void update(double pNow) {
+
+        mCurrentTime = pNow;
+
+        //Get encoder value from secondary Talon
+        mCurrentEncoderTicks = getEncoderPosition();
+        mDesiredDirectionUp = (mDesiredPower > 0);
+
+        updateElevator(pNow);
+        mDesiredPower = calculateDesiredPower(mCurrentState);
+        mDesiredPower *= 0.10d; //10% of the desired power
+
+        mMasterElevator.set(Util.limit(mDesiredPower, mMinPower, mMaxPower));
+
+    }
+
+    public void updateElevator(double pNow) {
+
+        //TODO test this
+        //Should override any automated control
+        if (Math.abs(mDesiredPower) ==  0d && !mAtBottom) { //Hold if input is absent and we aren't at the bottom
+            mCurrentState = EElevatorState.HOLD;
+        } else {
+            mCurrentState = EElevatorState.NORMAL;
+        }
+
+        if (mDesiredDirectionUp) {
+            shouldDecelerate(mCurrentEncoderTicks, mDesiredDirectionUp);
+        }
+
+    }
+
+    private void resetBottom() {
+        mCurrentEncoderTicks = 0;
+        zeroEncoder();
+    }
+
+    private void resetTop() {
+        setEncoderPosition(0); //Find out top position encoder ticks
+    }
+
+    public void zeroEncoder() {
+        setEncoderPosition(0); //Actually should be zero
+    }
+
+    //Checks if we are close enough to the top or bottom 
+    //that we should be decelerating
+    private void shouldDecelerate(int pCurrentEncoderTicks, boolean pCurrentDirectionUp) {
+        if (pCurrentDirectionUp) {
+            if (pCurrentEncoderTicks >= mUpEncoderThreshold) {
+                mCurrentState = EElevatorState.DECEL_TOP;
+            }
+        } else if (!pCurrentDirectionUp) {
+            if (pCurrentEncoderTicks >= mDownEncoderThreshold) {
+                mCurrentState = EElevatorState.DECEL_BOTTOM;
+            }
+        }
+    }
+    
+    public void toBottom() {
+        mDesiredPosition = EElevatorPosition.BOTTOM;
+        mCurrentState = EElevatorState.SET_POSITION;
+    }
+
+    public void toTop() {
+        mDesiredPosition = EElevatorPosition.BOTTOM;
+        mCurrentState = EElevatorState.SET_POSITION;
+    }
+
+
```

Comment:
This method isn't actually setting a position; it is instead returning a motor power based upon the current position and desired position.  It should be named and documented accordingly.